### PR TITLE
GPC-273: Notifications for admin actions

### PR DESCRIPTION
**Pull request by ethanmills**

### CODE REVIEW

**Comment by ethanmills** [comment]
Shout early if you're unhappy with the approach before I apply it everywhere

File: src/main/kotlin/uk/gov/gdx/datashare/services/AdminService.kt [hunk 1]
```diff
@@ -12,12 +12,14 @@ class AdminService(
...
   private val acquirersService: AcquirersService,
   private val suppliersService: SuppliersService,
+  private val adminActionAlertsService: AdminActionAlertsService,
 ) {
   fun createCognitoClient(cognitoClientRequest: CognitoClientRequest) =
     cognitoService.createUserPoolClient(cognitoClientRequest)
 
   @Transactional
   fun createAcquirer(createAcquirerRequest: CreateAcquirerRequest): CognitoClientResponse? {
+    adminActionAlertsService.noticeAction(AdminAction("Create Acquirer", createAcquirerRequest))
```
Comment (oskarwilliams378) [inline src/main/kotlin/uk/gov/gdx/datashare/services/AdminService.kt:22]: Would we want to alert on success or attempt? I fear that if this fails we would still get the alert saying that it had been created.

File: src/main/kotlin/uk/gov/gdx/datashare/services/AdminActionAlertsService.kt [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+import software.amazon.awssdk.services.sns.model.PublishRequest
+
+@Service
+class AdminActionAlertsService(
+  @Value("\${admin-action-alert-sns-topic-arn:#{null}}") private val topicArn: String?,
+  private val objectMapper: ObjectMapper,
+) {
+  private val snsClient by lazy { SnsClient.create() }
+
+  fun noticeAction(action: AdminAction) {
```
Comment (oskarwilliams378) [inline src/main/kotlin/uk/gov/gdx/datashare/services/AdminActionAlertsService.kt:18]: Is this the right method name?

**Review by oskarwilliams378 — COMMENTED** [review]
2 comments, but I like the overall approach

**Review by oskarwilliams378 — APPROVED** [review]

**Pull request merged**